### PR TITLE
configure add-on overview

### PR DESCRIPTION
**Pull request by kjaskiewiczz**

### CODE REVIEW

**Comment by kjaskiewiczz** [comment]
I know it's still WIP, but I wasn't sure what should be documented here;
@estenberg it would be great if you could take a brief look and possibly left some comment so I could adjust the docs;

File: 01.Get-started/chapter.md [hunk 1]
```diff
@@ -8,4 +8,4 @@ taxonomy:
 
 # Get started
 
-Quickly try the most common use cases of Mender.
+foo Quickly try the most common use cases of Mender.
```
Comment (merlin-northern) [inline 01.Get-started/chapter.md:11]: foo you say? ;-)

File: 09.Add-ons/02.Configure/docs.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+title: Configure
+taxonomy:
+    category: docs
+---
+
```
Comment (estenberg) [inline 09.Add-ons/02.Configure/docs.md:6]: Add a header about the commercial package, align with what @merlin-northern asked about for Troubleshoot add-ons.

E.g. "!! Configure is part of a commercial add-on package. See the [Mender features page](https://mender.io/plans/features?target=_blank) for an overview of all Mender plans and features."

File: 09.Add-ons/02.Configure/docs.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+![configuration-3](deviceconfig3.png)
+
+The configuration is applied in a similar way the updates are applied.
+The deviceconfig microservice which is responsible to manage the configuration
+uses the deployments service to update the configuration for the device.
+The picture below shows the configuration update flow.
+
+![configuration-4](configure1.png)
+
+The specialized update module is responsible for applying the configuration to the device.
```
Comment (estenberg) [inline 09.Add-ons/02.Configure/docs.md:21]: Link to the Update Module on Mender Hub (https://tracker.mender.io/browse/MEN-4424)

**Review by estenberg — CHANGES_REQUESTED** [review]
Added a few cosmetic comments.

Though I think we'll need a bit more here:

Architecture / how it works end-to-end
The screenshots and diagram provide a very high-level architecture & UI. But what do I do from here? How does it all tie together? Would at least have a header "How it works" and include the flow from when settings are typed in to the UI until they are applied & reported back to the backend.

Here is one diagram from planning: https://docs.google.com/presentation/d/1a7hQQDSxRJzMDWqjktFpMWrJ610MMruthwTpLx2IlFk/edit#slide=id.g6f2c5369fe_0_8
Here is another that includes the file paths (which I think is important); https://docs.google.com/presentation/d/1reuk8qu7aSsQf_IOzbYFa_JoLpWMwGDbCTQ98g4Y_6A/edit#slide=id.g9b313eab1b_0_228

Would take these as a starting point and create a diagram for how this all fits together, e.g. take the first diagram and add the file paths from the second, perhaps with a UI screenshot as well.

* Installation
How to install it is needed, similar to: https://docs.mender.io/add-ons/remote-terminal/installation. Even if the only thing needed is the Update Module, just write that in a new document and point to the Update Module on Mender Hub.

File: 09.Add-ons/02.Configure/docs.md (hunk 2 above)
Comment (kjaskiewiczz) [reply 09.Add-ons/02.Configure/docs.md:6]: I've added something; please take a look

File: 09.Add-ons/02.Configure/docs.md (hunk 3 above)
Comment (kjaskiewiczz) [reply 09.Add-ons/02.Configure/docs.md]: there is no info about this update module on the Mender Hub yet

File: 01.Get-started/chapter.md (hunk 1 above)
Comment (kjaskiewiczz) [reply 01.Get-started/chapter.md]: fixed

File: 09.Add-ons/02.Configure/docs.md (hunk 2 above)
Comment (estenberg) [reply 09.Add-ons/02.Configure/docs.md:6]: Text is fine, but try to style it the same as the other notes of this kind. E.g. take a look here, and steal the style (I think it's three exclamation marks): https://docs.mender.io/overview/delta-update

Comment (kjaskiewiczz) [reply 09.Add-ons/02.Configure/docs.md:6]: done

**Comment by estenberg** [comment]
It's quite good now, thanks for the improvements.

One thing I think still needs to be fixed is this configure1.png diagram: https://github.com/mendersoftware/mender-docs/blob/77c048a91cdbb16612de8d7b6baebcc4aba2b708/09.Add-ons/02.Configure/configure1.png

It's not quite how it turned out. For example, there is no state script involved AFIK? Should we include the file paths too? 

Save to: /var/lib/mender-configure/device-config.json
Execute (modified by end user): /usr/lib/mender-configure/apply-device-config.d/*

File: 09.Add-ons/02.Configure/docs.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+![configuration-3](deviceconfig3.png)
+
+# How it works
+
+The configuration is applied in a similar way the updates are applied.
+The deviceconfig microservice which is responsible to manage the configuration
+uses the deployments service to update the configuration for the device.
+The specialized update module is responsible for applying the configuration to the device.
+You can find more information about the configuration update module on [Mender Hub](https://hub.mender.io).
+The picture below shows the configuration update flow.
```
Comment (TheMeaningfulEngineer) [inline 09.Add-ons/02.Configure/docs.md:26]: FWIW
Separating the description between the server and client.

```suggestion
The configuration is applied in a similar way the updates are applied.
`deviceconfig` is a microservice on the server responsible for "translating" a configuration change made in the UI into a deployment.
It then proceeds to trigger a deployment to the device using the same mechanism used in OTA, the `deployments` microservice. 

On the device side, a specialized update module is responsible for applying the configuration.
You can find more information about the configuration update module on [Mender Hub](https://hub.mender.io).
The picture below shows the configuration update flow.
```

**Comment by TheMeaningfulEngineer** [comment]
@kjaskiewiczz 
How does the device provide the information to the `devicevonfig`. Is it pushing the updates at regular intervals or "real time"?

**Comment by kjaskiewiczz** [comment]
@TheMeaningfulEngineer the device is reporting back the configuration using this endpoint - https://staging.docs.mender.io/api/development/#report-device-configuration

**Comment by kjaskiewiczz** [comment]
> It's not quite how it turned out. For example, there is no state script involved AFIK? Should we include the file paths too?
@estenberg not sure if this should be a part of the overview; we are providing the detailed description of how the configure update module works in other place

**Comment by estenberg** [comment]
>> It's not quite how it turned out. For example, there is no state script involved AFIK? Should we include the file paths too?

> @estenberg not sure if this should be a part of the overview; we are providing the detailed description of how the configure update module works in other place

I understand, but I think at least a diagram that shows the flow is very helpful here. I think the file paths are helpful to understand it, but if you think it's too detailed then we can live without them here then. We still need the diagram to correctly reflect how it is though (now there are some errors, as I pointed out).

**Comment by kjaskiewiczz** [comment]
@estenberg  I've added paths to the diagram; please take a look

**Review by estenberg — APPROVED** [review]
Very good, thank you.

**Pull request merged**